features/filesystem/bd/ChainingBlockDevice.h

```diff
@@ -0,0 +1,153 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2012 ARM Limited
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#ifndef MBED_CHAINING_BLOCK_DEVICE_H
+#define MBED_CHAINING_BLOCK_DEVICE_H
+
+#include "BlockDevice.h"
+#include "mbed.h"
+
+
+/** Block device for chaining multiple block devices
+ *  with the similar block sizes at sequential addresses
+ *
+ *  @code
+ *  #include "mbed.h"
+ *  #include "HeapBlockDevice.h"
+ *  #include "ChainingBlockDevice.h"
+ *
+ *  // Create two smaller block devices with
+ *  // 64 and 32 blocks of size 512 bytes
+ *  HeapBlockDevice mem1(64*512, 512);
+ *  HeapBlockDevice mem2(32*512, 512);
+ *
+ *  // Create a block device backed by mem1 and mem2
+ *  // contains 96 blocks of size 512 bytes
+ *  BlockDevice *bds[] = {&mem1, &mem2};
+ *  ChainingBlockDevice chainmem(bds);
+ */
+class ChainingBlockDevice : public BlockDevice {
+public:
+    /** Lifetime of the memory block device
+     *
+     *  @param bds      Array of block devices to chain with sequential block addresses
+     *  @param count    Number of block devices to chain
+     *  @note All block devices must have the same block size
+     */
+    ChainingBlockDevice(BlockDevice **bds, size_t bd_count);
+
+    /** Lifetime of the memory block device
+     *
+     *  @param bds      Array of block devices to chain with sequential block addresses
+     *  @note All block devices must have the same block size
+     */
+    template <size_t SIZE>
```

Comment:
There's not a reason, I just wasn't sure if we had a standard style for template parameters. I can update to `Size`